compile/container.go

```diff
@@ -97,10 +97,7 @@ func (m *MapSpec) ForEachTypeReference(f func(TypeSpec) error) error {
 	if err := f(m.KeySpec); err != nil {
 		return err
 	}
-	if err := f(m.ValueSpec); err != nil {
-		return err
-	}
-	return nil
+	return f(m.ValueSpec)
```

Comment:
Again with the unrelated style changes ;)

Reply:
I had to make these changes to fix the lint errors!

Reply:
This is a lint error in Go 1.8 but not in 1.9. I changed Travis on this repo
to run for Go 1.9 only. This should no longer be a problem.